packages/query/src/bitcoin/bitcoin-client.ts

```diff
@@ -0,0 +1,292 @@
+import { HIRO_API_BASE_URL_MAINNET } from '@leather-wallet/constants';
+import axios from 'axios';
+
+import type { Paginated } from '../../types/api-types';
+import { useLeatherNetwork } from '../leather-query-provider';
+
+class Configuration {
+  constructor(public baseUrl: string) {}
+}
+
+export interface UtxoResponseItem {
+  txid: string;
+  vout: number;
+  status: {
+    confirmed: boolean;
+    block_height: number;
+    block_hash: string;
+    block_time: number;
+  };
+  value: number;
+}
+
+export interface UtxoWithDerivationPath extends UtxoResponseItem {
+  derivationPath: string;
+}
+
+interface BestinslotInscription {
```

Comment:
At some point we prob need to think better on where models/types live. Rn we have both a `models` and `types` package, and we have them living in individual packages like this. It will be hard to know where to find them?

Reply:
Good question, @kyranjamie what is the intention there vs the models pkg?

Reply:
In this example, the types describe the response of the best in slot API, as such believe they should remain co-located by the query.

Types/interfaces are language level constructs. I wouldn't group them all together, in the same way I wouldn't recommend grouping all React hooks together. It's not helpful for developers. If best in slot change their API, why should I have to dig through multiple packages to update them? When would I want to consume the bestinslot API separately from the query?

I don't think it's necessary to have both `models` (domain models) and `types` (Typescript types) packages in the definitions used here. `Money` is an `interface` that represents our domain modelling of money, therefore it should live in models. We don't really use class-based models, but if we did, they'd live there too. 

The types package in it's existing form is a package for external developers to easily add ts typings to their global object. Maybe we should call it something else, but the idea is it's extremely lightweight by nature of it being types-only. 

I've made a [database here documenting our packages, and their responsibilities](https://www.notion.so/trustmachines/ea186ca503cd4820bdc3ceb888930906?v=af67dc8559a94f69aa586465cbab4275&pvs=4). It's very important we get the design of these packages right first time, as changing them will be hard work. Adding packages should be deliberate and well-justified.